src/posix/tty.rs

```diff
@@ -82,6 +83,35 @@ impl TTYPort {
         Ok(port)
     }
 
+    pub fn list_ports() -> Vec<SerialPortInfo> {
```

Comment:
It doesn't seem right having this method on the device object. I think it's fine as a free function for now.